grafana-plugin/src/containers/UserSettings/parts/tabs/GoogleCalendar/GoogleCalendar.tsx

```diff
@@ -21,11 +21,16 @@ const GoogleCalendar: React.FC<{ id: ApiSchemas['User']['pk'] }> = observer(({ i
 
   const styles = useStyles2(getStyles);
 
-  const user = userStore.items[id];
-  const [googleCalendarSettings, setGoogleCalendarSettings] = useState(user?.google_calendar_settings);
-  const [showSchedulesDropdown, setShowSchedulesDropdown] = useState(
-    user.google_calendar_settings?.oncall_schedules_to_consider_for_shift_swaps?.length > 0
-  );
+  const [googleCalendarSettings, setGoogleCalendarSettings] =
+    useState<ApiSchemas['User']['google_calendar_settings']>();
+  const [showSchedulesDropdown, setShowSchedulesDropdown] = useState<boolean>();
+
+  useEffect(() => {
+    userStore.fetchItemById({ userPk: id }).then((user) => {
```

Comment:
note for @teodosii  and @brojd: we need to fetch user data directly from the component here because user object in the store may not contain `google_calendar_settings` data, because user data retrieved from `GET /users` endpoint does not contain `google_calendar_settings` object (this endpont uses other data serializer than `GET /user/<pk>`), let's discuss this problem together once Dominik is back

Reply:
There are probably more information exposed by `GET /user/<pk>` that `GET /users` lacks and that might be needed now or in the future in different parts of UserSettings (in different tabs / sections). GoogleCalendar is nested within one of the tabs of UserSettings. If any other tab needs information from `GET /user/<pk>`, with this approach we would need to replicate HTTP requests (which leads to things like [this one](https://github.com/grafana/oncall/issues/4112) that @joeyorlando brought up for a good reason). It would make sense to me to move this `useEffect` up in the components' structure - to UserSettings and then stop using local state for googleCalendarSettings as I mentioned in the previous PR.